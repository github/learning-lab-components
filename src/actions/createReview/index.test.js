const createReview = require('./index')
import mockContext from '../../tests/mockContext'

describe('createReview', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      pull_request: {
        number: 1
      }
    }, {
      pullRequests: {
        createReview: jest.fn()
      },
      search: {
        issues: jest.fn(() => Promise.resolve({
          data: {
            items: [{ title: 'My Pull Request', number: 3 }]
          }
        }))
      }
    })
  })

  it('creates a pull request review', async () => {
    await createReview(context, { event: 'APPROVED', body: 'a-body.md' })
    expect(context.github.pullRequests.createReview).toHaveBeenCalled()
    expect(context.github.pullRequests.createReview.mock.calls).toMatchSnapshot()
  })

  it('creates a pull request review with a number', async () => {
    await createReview(context, { event: 'APPROVED', body: 'a-body.md', number: 2 })
    expect(context.github.pullRequests.createReview).toHaveBeenCalled()
    expect(context.github.pullRequests.createReview.mock.calls).toMatchSnapshot()
  })

  it('creates a pull request review with pullRequest title field', async () => {
    await createReview(context, { event: 'APPROVED', body: 'a-body.md', pullRequest: 'My Pull Request' })
    expect(context.github.pullRequests.createReview).toHaveBeenCalled()
    expect(context.github.pullRequests.createReview.mock.calls).toMatchSnapshot()
  })

  it('creates a pull request review with pullRequest number field', async () => {
    await createReview(context, { event: 'APPROVED', body: 'a-body.md', pullRequest: 4 })
    expect(context.github.pullRequests.createReview).toHaveBeenCalled()
    expect(context.github.pullRequests.createReview.mock.calls).toMatchSnapshot()
  })

  it('uses the number when hardcoded', async () => {
    await createReview(context, { event: 'APPROVED', body: 'a-body.md', pullRequest: 'My Pull Request', number: 1000 })
    expect(context.github.pullRequests.createReview).toHaveBeenCalled()
    expect(context.github.pullRequests.createReview.mock.calls).toMatchSnapshot()
  })
})
