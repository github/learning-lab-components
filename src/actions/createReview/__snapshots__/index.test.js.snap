// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createReview creates a pull request review 1`] = `
Array [
  Array [
    Object {
      "body": "a-body.md",
      "event": "APPROVED",
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`createReview creates a pull request review with a number 1`] = `
Array [
  Array [
    Object {
      "body": "a-body.md",
      "event": "APPROVED",
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`createReview creates a pull request review with pullRequest number field 1`] = `
Array [
  Array [
    Object {
      "body": "a-body.md",
      "event": "APPROVED",
      "number": 4,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`createReview creates a pull request review with pullRequest title field 1`] = `
Array [
  Array [
    Object {
      "body": "a-body.md",
      "event": "APPROVED",
      "number": 3,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`createReview uses the number when hardcoded 1`] = `
Array [
  Array [
    Object {
      "body": "a-body.md",
      "event": "APPROVED",
      "number": 1000,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;
